src/python/turicreate/toolkits/object_detector/_tf_model_architecture.py

```diff
@@ -380,7 +380,7 @@ def train(self, feed_dict):
         """
         for key in feed_dict.keys():
             feed_dict[key] = _utils.convert_shared_float_array_to_numpy(feed_dict[key])
-        feed_dict['labels'] = feed_dict['labels'].reshape(self.batch_size, self.grid_shape[0], self.grid_shape[1],self.num_anchors, self.num_classes + 5)
+        feed_dict['labels'] = feed_dict['labels'].reshape(-1, self.grid_shape[0], self.grid_shape[1],self.num_anchors, self.num_classes + 5)
```

Comment:
not sure why here is not using `None`. what does `-1` mean? It's better if you add a line comment.

Reply:
If we set the dimension to -1, the number will be automatically inferred.
Say you have a matrix which size [2,6]
if you reshape it to [-1,4],
the first element will be 3.

Reply:
Yeah, could we add a comment explaining this line?

Reply:
great. just make sure the behavior is consistent. You can use -1 for placeholder too if you think that's a trivial change to make.